src/plugins/intel_gna/tests/functional/limitations/layers_limit.cpp

```diff
@@ -40,34 +40,34 @@ class GNALayersLimitTest : public testing::WithParamInterface<GNALayersLimitTest
 protected:
     void SetUp() override {
         ov::element::Type net_type = ov::element::f32;
-        auto params = ngraph::builder::makeParams(net_type, {{1, 2}, {1, 2}});
+        const size_t branch_count = 64;
+        std::vector<std::vector<size_t>> shapes(branch_count, {1, 2});
+
         std::map<std::string, std::string> common_conf, conf;
         size_t layers_number;
         std::tie(targetDevice, common_conf, conf, layers_number) = this->GetParam();
 
         configuration.insert(common_conf.begin(), common_conf.end());
         configuration.insert(conf.begin(), conf.end());
 
-        auto add_const = ngraph::builder::makeConstant(net_type, ngraph::Shape{1}, std::vector<float>{0.01f});
-        auto add_x = std::make_shared<ngraph::opset8::Add>(add_const, params[0]);
-        auto add_y = std::make_shared<ngraph::opset8::Add>(add_const, params[1]);
+        auto params = ngraph::builder::makeParams(net_type, shapes);
+        auto add_const = ngraph::builder::makeConstant(net_type, ov::Shape{1}, std::vector<float>{0.01f});
+        ov::ResultVector results;
 
-        std::vector<std::shared_ptr<ov::op::v1::Add>> add_nodes_x;
-        std::vector<std::shared_ptr<ov::op::v1::Add>> add_nodes_y;
+        std::vector<std::vector<std::shared_ptr<Add>>> branches;
 
-        add_nodes_x.push_back(add_x);
-        add_nodes_y.push_back(add_y);
+        for (size_t i = 0; i < branch_count; ++i) {
+            configuration.insert({"GNA_SCALE_FACTOR_" + std::to_string(i), "1"});
+            std::vector<std::shared_ptr<Add>> add_nodes;
+            add_nodes.push_back(std::make_shared<Add>(add_const, params[i]));
 
-        for (size_t i = 0; i < (layers_number - 2) / 2; ++i) {
-            auto add_next_x = std::make_shared<ngraph::opset8::Add>(add_nodes_x.back(), params[0]);
-            auto add_next_y = std::make_shared<ngraph::opset8::Add>(add_nodes_y.back(), params[1]);
-            add_nodes_x.push_back(add_next_x);
-            add_nodes_y.push_back(add_next_y);
+            for (size_t j = 0; j < (layers_number - branch_count) / branch_count; ++j) {
+                add_nodes.push_back(std::make_shared<Add>(add_nodes.back(), params[i]));
```

Comment:
are you sure that number of final layers in Gna2Model will be the same as in input params?